### PR TITLE
Bugfix: fix nodeProto._setSiblingIndex

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos/3d-tasks/issues/16288

### Changelog

* Bugfix: node.jsb.ts  Node.setSiblingIndex

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by minggo — APPROVED** [review]

File: native/cocos/bindings/manual/jsb_scene_manual.cpp [hunk 1]
```diff
@@ -620,6 +620,29 @@ static bool js_scene_Node_inverseTransformPoint(void *nativeObject) // NOLINT(re
...
+        return false;
+    }
+    if (!args[0].isObject() || !args[0].toObject()->isArray()) {
+        SE_REPORT_ERROR("wrong argument type, array expected");
+        return false;
+    }
+
+    se::Object *childrenObj = args[0].toObject();
+    uint32_t len;
+    childrenObj->getArrayLength(&len);
```
Comment (dumganhar) [inline native/cocos/bindings/manual/jsb_scene_manual.cpp]: What's `len` variable used for ?

Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_scene_manual.cpp:638]: ```suggestion```

Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_scene_manual.cpp]:
```suggestion
```

Comment (PatriceJiang) [reply native/cocos/bindings/manual/jsb_scene_manual.cpp]: removed

**Review by dumganhar — APPROVED** [review]

**Pull request merged**